hydra-node/src/Hydra/Network/Ouroboros.hs

```diff
@@ -167,24 +211,36 @@ withOuroborosNetwork tracer localHost remoteHosts networkCallback between = do
       , spSubscriptionTarget = IPSubscriptionTarget remoteAddrs (length remoteAddrs)
       }
 
+  actualConnect ::
+    IOManager ->
+    IO t ->
+    (t -> OuroborosApplicationWithMinimalCtx 'InitiatorMode SockAddr LByteString IO () Void) ->
+    Socket ->
+    IO ()
   actualConnect iomgr newBroadcastChannel app sn = do
     chan <- newBroadcastChannel
     connectToNodeSocket
       iomgr
-      unversionedHandshakeCodec
+      (codecHandshake versionNumberCodec)
       noTimeLimitsHandshake
-      unversionedProtocolDataCodec
+      (cborTermVersionDataCodec dataCodecCBORTerm)
```

Comment:
Do we need this? As we will only support one version, the handshake should already identify whether we have a compatible version. AFAIS this allows for versioned data encoding/decoding. That is, support multiple versions.